HGustavs/LenaSYS
# Added ability to restart the installer from specified step

The installer now creates an associative array containing all the operations the installer needs to perform. This array is constructed based on the settings provided via json from the frontend. Specifying the parameter: `starting_step: "add_test_data"` will for example restart the installer from the add_test_data step. 

# How to test?
1. First change the file `LenaSYS/newinstaller/tools/install_engine.php` to contain the proper pdo authentication settings. 
![image](https://github.com/HGustavs/LenaSYS/assets/129264968/2ab7c8d7-f85a-4cec-b8c5-890ee43a6df8)



2. Since everything should be working correctly, (assuming it does, thats why you are testing haha) you first have to get it to fail somewhere. This can be achieved by changing the installer options in the file `LenaSYS/newinstaller/installer_ui.php`. For example, change the language support list to a language that does not exist. This should result in a fail, where it tells you where it failed. 
![image](https://github.com/HGustavs/LenaSYS/assets/129264968/4da5fe62-e91a-4183-8c5d-df778c01cdd0)
Running the code by navigating to [localhost/LenaSYS/newinstaller/installer.php](http://localhost/LenaSYS/newinstaller/installer.php) should now result in the following exception.
![image](https://github.com/HGustavs/LenaSYS/assets/129264968/219050e5-5885-44ca-9cde-418910e4f471)

3. Now remove the language that you added that does not exist. And set the starting step to `add_language_support_html`. Should look like this:
![image](https://github.com/HGustavs/LenaSYS/assets/129264968/4f562a5d-cfa2-4ac7-aa66-ea58644881a4)
And give the following output.
![image](https://github.com/HGustavs/LenaSYS/assets/129264968/25ee2de0-97a5-40e1-bf64-6e5d0dae2cc6)
Notice how the first step now was to install the html language keywords.
